delete text

alinagazitdinova opened the pull request.

alinagazitdinova committed:
delete text

alinagazitdinova committed:
add project

egorvas commented on src/test/java/samokat/OrderCheck.java:
```diff
@@ -0,0 +1,125 @@
+        MainPageOrder mainPageOrder = new MainPageOrder(driver);
+
+        mainPageOrder.open(); //Открыть страницу
+        mainPageOrder.clickOrderButton(); // Нажать на кнопку Заказать
```
У тебя два фактически одинаковых теста, рекомендую сделать один, а кнопку заказа внести в параметр

egorvas commented on src/test/java/samokat/ImportantQuestionsCheckText.java:
```diff
@@ -0,0 +1,177 @@
+        mainPageSeven.scrollPage();
+
+        mainPageSeven.clickHeading8();
+        Thread.sleep(500);
```
Тут можно без слипа

egorvas commented on src/test/java/samokat/ImportantQuestionsCheckText.java:
```diff
@@ -0,0 +1,177 @@
+    }
+
+    @Test
+    public void neededTextWhenClicking1() throws InterruptedException {
```
По сути у тебя куча однотипных элементов, тут отлично зашла бы параметризация.

egorvas commented on src/main/java/pom/MainPageFour.java:
```diff
@@ -0,0 +1,29 @@
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+
+public class MainPageFour {
```
Я совершенно не понимаю твой подход. У тебя физически одна главная страница, а ты зачем-то сделала кучу PO MainPageNumber.
Просто внести все локаторы и функции в один PO и тебе не нужно будет заниматься копипастой по созданию классов.

egorvas commented on src/main/java/pom/OrderForm1.java:
```diff
@@ -0,0 +1,57 @@
+import org.openqa.selenium.WebDriver;
+
+public class OrderForm1 {
+    public final By userName = By.xpath(".//*[@placeholder='* Имя']");
```
Обрати внимание, что по условию задания нужно выписать комментарии к каждому локатору: "Выпиши все элементы в список с помощью комментариев..."

alinagazitdinova committed:
Изменения:
1) Кнопка Заказать внесена в параметризацию.
2) Добавлена параметризация в тест neededTextWhenClicking
3) Убрала ненужные MainPageNumber
4) Добавила описание локаторов в комментариях
5) Sleep не получилось убрать, так как без них периодически некоторые тесты падают, в комментариях описала другие возожные варианты, которые пробовала.

egorvas commented on src/test/java/samokat/ImportantQuestionsCheckText.java:
```diff
@@ -0,0 +1,71 @@
+    @Parameterized.Parameters
+    public static Object[][] getData() {
+        return new Object[][] {
+                {"accordion__heading-0", "//*[@id= 'accordion__panel-0']/p", "Сутки — 400 рублей. Оплата курьеру — наличными или картой."},
```
Все таки локаторы лучше хранить в PO, а в тесте использовать уже объекты
